parse_node/parse_source_file.ts

```diff
@@ -21,18 +28,78 @@ const getClassDeclarationHeader = (
 
     const clause = node.heritageClauses[0] as ts.HeritageClause
     const type = clause.types[0]
+    const tsType = props.program.getTypeChecker().getTypeAtLocation(type)
 
     extendsFrom = type.getText()
+
+    if (tsType.symbol && tsType.symbol.declarations) {
+      // Handle the case when extended class is a inner clas of file
+      const classDecl = tsType.symbol.declarations.find((v) =>
+        ts.isClassDeclaration(v)
+      ) as ClassDeclaration | null
+
+      if (classDecl) {
+        const modifiers = (classDecl.modifiers ?? []).map((v) => v.getText())
+
+        const asset = props.project
+          .sourceFiles()
+          .find((v) => v.fsPath === classDecl.getSourceFile().fileName)
+
+        if (
+          !asset &&
+          !modifiers.includes("declare") &&
+          (!modifiers.includes("default") || !classDecl.name)
+        ) {
+          addError({
+            description:
+              "Class extends a type for which a source file is missing. This is an internal ts2gd bug. Please report it.",
```

Comment:
I like naming exactly what's going wrong in the error. e.g. I would rewrite this like so:

`Class ${ className } extends ${ nameWeCantFind } for which...`

Reply:
Done